### ISSUE TITLE
IndexOutOfBoundsException during playback

### ISSUE THREAD

**Issue by kkovach** [google/ExoPlayer#6776]
### [REQUIRED] Issue description
I get the following exception and crash while playing audio.

```
Fatal Exception: java.lang.IndexOutOfBoundsException
       at com.google.android.exoplayer2.util.Assertions.checkIndex(Assertions.java:68)
       at com.google.android.exoplayer2.source.SinglePeriodTimeline.getWindow(SinglePeriodTimeline.java:175)
       at com.google.android.exoplayer2.source.MaskingMediaSource$MaskingTimeline.getWindow(MaskingMediaSource.java:266)
       at com.google.android.exoplayer2.source.AbstractConcatenatedTimeline.getWindow(AbstractConcatenatedTimeline.java:199)
       at com.google.android.exoplayer2.Timeline.getWindow(Timeline.java:668)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.generateEventTime(AnalyticsCollector.java:612)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.generateEventTime(AnalyticsCollector.java:636)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.generateMediaPeriodEventTime(AnalyticsCollector.java:662)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.onMediaPeriodCreated(AnalyticsCollector.java:318)
       at com.google.android.exoplayer2.source.MediaSourceEventListener$EventDispatcher.lambda$mediaPeriodCreated$0$MediaSourceEventListener$EventDispatcher(MediaSourceEventListener.java:371)
       at com.google.android.exoplayer2.source.-$$Lambda$MediaSourceEventListener$EventDispatcher$N-EOPAK5UK0--YMNjezq7UM3UNI.run(:6)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:214)
       at android.os.HandlerThread.run(HandlerThread.java:67)
```

### [REQUIRED] Reproduction steps
I cannot reliably reproduce this issue, but I suspect that it might be caused by multiple calls to ConcatenatingMediaSource.addMediaSouce() Rough application flow is as follows...

- Add single MediaSource to a ConcatenatingMediaSource
- Prepare player with that ConcatenatingMediaSource
- Once the timeline has changed set speed and playWhenReady to true
- Add multiple (variable number of) additional MediaSources to the ConcatenatingMediaSource

### [REQUIRED] Link to test content
Do not have any of this right now.

### [REQUIRED] A full bug report captured from the device
Will do if necessary.

### [REQUIRED] Version of ExoPlayer being used
2.11.0

### [REQUIRED] Device(s) and version(s) of Android being used
Pixel 3 - Android 10

**Comment by Samrobbo** [comment]
Hi,

We'll look into this in the new year, as the team are almost all taking time off for the holiday period.

**Comment by tonihei** [comment]
This issue was already fixed by a [recent change](https://github.com/google/ExoPlayer/commit/36fa9d5a434812ef637ad27a0a9f1c4dd9651b1b#). 

This change may still be some way away from being part of a release. I added a [smaller fix to 2.11.1](https://github.com/google/ExoPlayer/commit/5d728abc41cc7fd893a90083653baddfec3275ed) too
